File: assignment_1.py
grandmas_response = {
    "empty_input": "WHAT?!",
    "lowercase_input": "SPEAK UP, KID!",
    "uppercase_input": "NO, NOT SINCE 1946!",
    "first_goodbye": "LEAVING SO SOON?",
    "second_goodbye": "LATER, SKATER!",
}


def deaf_grandma() -> None:
    goodbye_count = 0

    while True:
        user_input = input("Speak to grandma: \n")

        if not user_input:
            print(grandmas_response["empty_input"], "\n")
        elif user_input == "GOODBYE!":
            goodbye_count += 1
            if goodbye_count == 1:
                print(grandmas_response["first_goodbye"], "\n")
            elif goodbye_count == 2:
                print(grandmas_response["second_goodbye"], "\n")
                break
        elif any(char.islower() for char in user_input):
            print(grandmas_response["lowercase_input"], "\n")
        elif user_input.isupper():
            print(grandmas_response["uppercase_input"], "\n")

deaf_grandma()

File: assignment_1.js
const readline = require("readline");

function deafGrandma() {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  const responses = {
    emptyInput: "WHAT?!",
    lowercaseInput: "SPEAK UP, KID!",
    uppercaseInput: "NO, NOT SINCE 1946!",
    firstGoodbye: "LEAVING SO SOON?",
    secondGoodbye: "LATER, SKATER!",
  };

  let goodbyeCount = 0;

  const askGrandma = () => {
    const regexLower = /[a-z]/;
    const regexUpper = /[A-Z]/;
    rl.question("Speak to grandma: ", (userInput) => {
      if (userInput.trim() === "") {
        console.log(responses.emptyInput);
      } else if (userInput === "GOODBYE!") {
        goodbyeCount++;
        if (goodbyeCount == 1) {
          console.log(responses.firstGoodbye);
        } else if (goodbyeCount == 2) {
          console.log(responses.secondGoodbye);
          rl.close();
          return;
        }
      } else if (regexLower.test(userInput)) {
        console.log(responses.lowercaseInput);
      } else if (regexUpper.test(userInput)) {
        console.log(responses.uppercaseInput);
      }
      askGrandma();
    });
  };
  askGrandma();
}

deafGrandma();
